source/crud/tests/README.md

```diff
@@ -201,3 +201,70 @@ CommandSucceededEvents. Then, insert an invalid document (e.g. `{x: 1}`) and ass
 code is `121` (i.e. DocumentValidationFailure), and that its `details` property is accessible. Additionally, assert that
 a CommandSucceededEvent was observed and that the `writeErrors[0].errInfo` field in the response document matches the
 WriteError's `details` property.
+
+### 3. `MongoClient.bulkWrite` batch splits a `writeModels` input with greater than `maxWriteBatchSize` operations
+
+Test that `MongoClient.bulkWrite` properly handles `writeModels` inputs containing a number of writes greater than
+`maxWriteBatchSize`.
+
+This test must only be run on 8.0+ servers.
+
+Construct a `MongoClient` (referred to as `client`) with
+[command monitoring](../../command-logging-and-monitoring/command-logging-and-monitoring.rst) enabled to observe
+CommandStartedEvents. Perform a `hello` command using `client` and record the `maxWriteBatchSize` value contained in the
+response. Then, construct the following write model (referred to as `writeModel`):
+
+```json
+InsertOne: {
+  "namespace": "db.coll",
+  "document": { "a": "b" }
+}
+```
+
+Construct a list of write models (referred to as `models`) with `writeModel` repeated `maxWriteBatchSize + 1` times.
+Execute `bulkWrite` on `client` with `models`. Assert that the bulk write succeeds and returns a `BulkWriteResult`
+with an `insertedCount` value of `maxWriteBatchSize + 1`.
+
+Assert that two CommandStartedEvents (referred to as `firstEvent` and `secondEvent`) were observed for the `bulkWrite` command.
+Assert that the length of `firstEvent.command.ops` is `maxWriteBatchSize`. Assert that the length of `secondEvent.command.ops`
+is 1. If the driver exposes `operationId`s in its CommandStartedEvents, assert that `firstEvent.operationId` is equal to
+`secondEvent.operationId`.
+
+### 4. `MongoClient.bulkWrite` batch splits when an `ops` payload exceeds `maxMessageSizeBytes`
+
+Test that `MongoClient.bulkWrite` properly handles a `writeModels` input which constructs an `ops` array larger than
+`maxMessageSizeBytes`.
+
+This test must only be run on 8.0+ servers.
+
+Construct a `MongoClient` (referred to as `client`) with
+[command monitoring](../../command-logging-and-monitoring/command-logging-and-monitoring.rst) enabled to observe
+CommandStartedEvents. Perform a `hello` command using `client` and record the following values from the response:
+`maxBsonObjectSize` and `maxMessageSizeBytes`. Then, construct the following document (referred to as `document`):
+
+```json
+{
+  "a": "b".repeat(maxBsonObjectSize - 500)
+}
+```
+
+Construct the following write model (referred to as `model`):
+
+```json
+InsertOne: {
+  "namespace": "db.coll",
+  "document": document
+}
+```
+
```

Comment:
Suggest testing the maximum in one batch to ensure drivers are not splitting earlier than expected.


```suggestion

#### Test maximum in one batch

Use the following calculation to determine the number of inserts that should be provided to `MongoClient.bulkWrite`:
`maxMessageSizeBytes / maxBsonObjectSize` (referred to as `numModels`). This number is intended to be close
to the maximum before splitting.

Construct as list of write models (referred to as `models`) with `model` repeated `numModels` times. Then execute
`bulkWrite` on `client` with `models`. Assert that the bulk write succeeds and returns a `BulkWriteResult` with
an `insertedCount` value of `numModels`.

Assert that one CommandStartedEvents (referred to as `firstEvent`) was observed. Assert that the length of
`firstEvent.command.ops` is equal to `numModels`.

#### Test two batches
```

Reply:
Good point. Retracting suggestion.